### ISSUE TITLE
Accessing User Created Folders Contents

### ISSUE THREAD

**Issue by kraulin** [ecederstrand/exchangelib#30]
I'm trying to access a folder inside my account called 'Undeliverable' and get a list of the emails inside and see if they have attachments.  

I've successfully retrieved a list of all the folders in my account and 'Undeliverable' is listed in the response.

I did a:
```
for i in account.folders:
...     print(i)
```
[folderIterations.txt](https://github.com/ecederstrand/exchangelib/files/561952/folderIterations.txt)

Now, that code shows <class 'exchangelib.folders.Root'>.  that's where all the folders I've created live.
How do I get inside the array that Root contains to then do something akin to:
account.inbox.Root['Undeliverable].find_items() and get all the data associated with that email, attachments and all?

I've done some investigation in the python prompt but I can't seem to getr inside the folder designated Root.

Currently, This is as far as I've gotten:

```
from exchangelib import DELEGATE
from exchangelib.account import Account
from exchangelib.configuration import Configuration
from exchangelib.services import IdOnly

from exchangelib import DELEGATE, IMPERSONATION, Account, Credentials, EWSDateTime, EWSTimeZone, Configuration, NTLM, CalendarItem, Q

from exchangelib.folders import Folder

credentials = Credentials(username='XXXXXX\XXXXXX', password='XXXXXX')
account = Account(primary_smtp_address='XXX@XXX', credentials=credentials, autodiscover=True, access_type=DELEGATE)

# Get Exchange ID and changekey of all inbox items
ids = account.inbox.find_items(shape=IdOnly)

# Get the rest of the attributes on the items
items = account.inbox.get_items(ids)

print(len(items))

for i in account.folders:
	print(i)
```

Thanks,
Bryan James

**Comment by ecederstrand** [comment]
Getting to the folder named 'Undeliverable' is a little bit clunky. You need to do this:

```
my_folder = None
for k, v in account.folders.items():
    for f in v:
        if f.name == 'Undeliverable':
            my_folder = f
            break
    if my_folder:
        break

assert my_folder
```

If you already know the folder class of your folder, e.g. `Messages`, you can shorten this to:

```
for f in account.folders[Messages]:
        if f.name == 'Undeliverable':
            my_folder = f
            break
else:
    raise Exception('Folder not found')
```

There should probably be an easier way to do this.

**Comment by ecederstrand** [comment]
I just committed a change so you can do:

`my_folder = account.root.get_folder_by_name('Undeliverable')
`

Change `root` to another folder if 'Undeliverable' is not located under `root`.

**Comment by MartinThoma** [comment]
> for f in account.folders[Messages]:

Where does `Messages` come from? I tried `from exchangelib import Messages` but it fails.

**Comment by ecederstrand** [comment]
I've thought about this before, but then it would only make sense to import all the other folders types, and that would pollute the top-level import quite a bit. So for this atypical use case I think it's ok to do `from exchangelib.folders import Message`.

Anyway, the `account.folders` dict feels like a hack. I'd prefer to find a better solution to accessing the folder structure in #110
